frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSquashAction.java

```diff
@@ -0,0 +1,173 @@
+import git4idea.repo.GitRepository;
+import io.vavr.Tuple2;
+import io.vavr.collection.List;
+import io.vavr.control.Option;
+import kotlin.Unit;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.backend.api.ICommitOfManagedBranch;
+import com.virtuslab.gitmachete.frontend.actions.common.VcsCommitMetadataAdapter;
+import com.virtuslab.gitmachete.frontend.actions.contextmenu.CheckoutSelectedBranchAction;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.GitNewCommitMessageActionDialog;
+import com.virtuslab.gitmachete.frontend.defs.ActionPlaces;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod(GitMacheteBundle.class)
+@CustomLog
+public abstract class BaseSquashAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IBranchNameProvider {
+
+  private final String LS = System.lineSeparator();
+
+  @Override
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+
+    Presentation presentation = anActionEvent.getPresentation();
+
+    val branchName = getNameOfBranchUnderAction(anActionEvent);
+    val nonRootBranchOption = branchName.flatMap(bn -> getManagedBranchByName(anActionEvent, bn))
+        .filter(b -> b.isNonRoot())
+        .map(b -> {
+          assert b.isNonRoot() : "branch is root";
+          return b.asNonRoot();
+        });
+    val syncToParentStatus = nonRootBranchOption.map(b -> b.getSyncToParentStatus()).getOrNull();
+    val numberOfCommitsOption = nonRootBranchOption.map(b -> b.getCommits().length());
+
+    if (branchName.isDefined() && nonRootBranchOption.isEmpty()) {
+      presentation.setDescription(
+          getNonHtmlString("action.GitMachete.BaseSquashAction.branch-is-root")
+              .format(branchName.get()));
+      presentation.setEnabled(false);
+
+    } else if (branchName.isDefined() && numberOfCommitsOption.isDefined()) {
+
+      val numberOfCommits = numberOfCommitsOption.getOrNull();
+      assert numberOfCommits != null : "unknown number of commits";
+
+      if (numberOfCommits < 2) {
+        presentation.setDescription(
+            getNonHtmlString("action.GitMachete.BaseSquashAction.not-enough-commits")
+                .format(branchName.get(), numberOfCommits.toString()));
+        presentation.setEnabled(false);
+      } else if (syncToParentStatus == InSyncButForkPointOff) {
+        presentation.setEnabled(false);
+        presentation.setDescription(getNonHtmlString("action.GitMachete.BaseSquashAction.fork-point-off")
+            .format(branchName.get()));
+      } else {
+        val isSquashingCurrent = getCurrentBranchNameIfManaged(anActionEvent)
```

Comment:
`isSquashingCurrentBranch`? coz the current name suggests sth like _is squashing in progress_